moped-toolbox/amd_milestones_backfill/queries.py

```diff
@@ -0,0 +1,64 @@
+"""
+Query projects which:
+- are not deleted
+- are not in a completed or post-construction phase
+- have a project type that matches the (currently) four project types which support
+  templates: Signal - Mod, Signal - New, PHB - Mod, PHB - New
+- were created on or prior to a user-supplied date ($max_date_added)
+"""
+PROJECTS_QUERY = """query NeedsMilestonesProjects($max_date_added: timestamptz!) {
+	moped_project(
+		order_by: { project_id: asc }
+		where: {
+			_and: [
+				# not-deleted projects only
+				{ is_deleted: { _eq: false } }
+				# exclude completed and post-construction projects
+				{
+					_and: [
+						{ current_phase: { _nilike: "complete" } }
+						{ current_phase: { _nilike: "post-construction" } }
+					]
+				}
+				# must contain at least one not-deleted project type related to signals or PHBs
+				{
+					moped_project_types: {
+						_and: [
+							{ moped_type: { type_name: { _similar: "%(Signal|PHB)%" } } }
+							{ status_id: { _eq: 1 } }
```

Comment:
while testing against staging, i uncovered a bug stemming from the fact i overlooked that there are "deleted" `moped_project_types` in the DB. i added `{ status_id: { _eq: 1 } }` here to ensure we ignored those.